File: src/dotnet/Ansight/Pairing/Models/ConnectRequest.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Handshake request sent to the host before a live pairing WebSocket session is opened.
/// </summary>
public sealed class ConnectRequest
{
    /// <summary>
    /// Protocol message type identifier.
    /// </summary>
    public required string Type { get; set; }

    /// <summary>
    /// Handshake protocol version.
    /// </summary>
    public required int Ver { get; set; }

    /// <summary>
    /// Unique identifier of the pairing config being used.
    /// </summary>
    public required string ConfigId { get; set; }

    /// <summary>
    /// One-time token that authorizes the connection attempt.
    /// </summary>
    public required string OneTimeToken { get; set; }

    /// <summary>
    /// App identifier expected by the pairing config.
    /// </summary>
    public required string AppId { get; set; }

    /// <summary>
    /// Human-readable name presented to the host for this client.
    /// </summary>
    public required string ClientName { get; set; }

    /// <summary>
    /// Process-lifetime identifier for the SDK runtime instance.
    /// </summary>
    public string? ProcessSessionId { get; set; }

    /// <summary>
    /// Indicates that the request came from a build-time developer pairing marker rather than a signed pairing config.
    /// </summary>
    public bool DevelopmentPairing { get; set; }
}

File: src/ios/Sources/AnsightKit/ConnectRequest.swift
import Foundation
import Network

public struct ConnectRequest: Sendable, Codable, Equatable {
    public let type: String
    public let ver: Int
    public let configId: String
    public let oneTimeToken: String
    public let appId: String
    public let clientName: String
    public let processSessionId: String?

    public init(
        type: String = "CONNECT_REQ",
        ver: Int = 1,
        configId: String,
        oneTimeToken: String,
        appId: String,
        clientName: String,
        processSessionId: String? = ProcessSessionIdentity.current
    ) {
        self.type = type
        self.ver = ver
        self.configId = configId
        self.oneTimeToken = oneTimeToken
        self.appId = appId
        self.clientName = clientName
        self.processSessionId = processSessionId
    }
}
